builds/checkin/dotnet.yaml

```diff
@@ -4,18 +4,32 @@ pr:
     include:
       - master
       - release/*
-  paths:
-    exclude:
-      - edgelet/*
-      - doc/*
 jobs:
 
+################################################################################
+  - job: check_run_pipeline
+################################################################################
+    displayName: Check pipeline preconditions (changes ARE NOT in edgelet and docs)
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i -v '^(edgelet|doc)'
+          if [[ $? == 0 ]]; then
+            echo "Detected changes outside of edgelet and docs folders"
+            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
```

Comment:
nit:Please use camel case for variable name in pipeline.

Reply:
but everywhere else we use UPPER_CASE, no?

Reply:
It is different from other yamls. AFAIK, we use all uppercase for environment variable. Normally I use lowercase as predefined variable is camel. Let’s confirm with @Damon what should be. I can update other yamls.

Reply:
I just did a quick survey of the source code. For instances of `task.setvariable`, it looks like we use UPPER_SNAKE_CASE consistently everywhere...except in the `e2e*.yaml` files I created 6 months ago. 😃

For `variables:`, I see **a lot** more variation. It's mostly dotted lowercase, but I see some UPPER_SNAKE_CASE, camelCase, and PascalCase too.

For `inputs:`, I see mostly camelCase, but there's some PascalCase and UPPER_SNAKE_CASE as well.

We should probably come up with a consistent standard; the wrinkle is that these variables get used in lots of different environments (bash, PowerShell, Azure pipelines logic, etc.), so it's natural to use the environment's standard. But in cases where variables are used in two or more environments with different conventions, which do you pick?